### ISSUE TITLE
Rules don't work for enum types (used to in v < 1)

### ISSUE THREAD

**Issue by alexandru-calinoiu** [dry-rb/dry-validation#561]
Summary of what the feature is supposed to do.

## Examples

Given this type:

```ruby
module Types
  include Dry.Types()

  RotationPattern = Types::Strict::String.enum('Weekly', 'Bi-Weekly')
end
```

And the following contract:

```ruby
class CreateContract < Api::Schema
  params do
    required(:rotation_pattern).filled(Types::RotationPattern)
  end
end
```

The following code should raise a validation error:

```ruby
CreateContract.new.call(rotation_pattern: 'Every now and then')
```

It used to raise an error `rotation_pattern: ['must be one of: Weekly, Bi-Weekly']` it does not work anymore

## Resources

none

## Workaround

As a work around we can change the rule to:

```ruby
          required(:rotation_pattern).filled(:str?, included_in?: Types::RotationPattern.values)
```

**Comment by flash-gordon** [comment]
This is addressed in https://github.com/dry-rb/dry-schema/pull/164 new version will soon be released.
